tests/rados/test_node_drain_customer_bug.py

```diff
@@ -0,0 +1,271 @@
+    service_obj = ServiceabilityMethods(cluster=ceph_cluster, **config)
+    ceph_nodes = kw.get("ceph_nodes")
+    config = kw["config"]
+
+    replicated_config = config.get("replicated_pool")
+    pool_name = replicated_config["pool_name"]
+
+    if not rados_obj.create_pool(pool_name=pool_name):
+        log.error("Failed to create the  Pool")
+        return 1
+
+    rados_obj.bench_write(pool_name=pool_name, byte_size="5M", rados_write_duration=60)
+    mgr_daemon = Thread(
+        target=background_mgr_task, kwargs={"mgr_object": mgr_obj}, daemon=True
+    )
+    # Printing the hosts in cluster
+    cmd_host_ls = "ceph orch host ls"
+    out = rados_obj.run_ceph_command(cmd=cmd_host_ls)
+    log.info(f"The hosts in the cluster before starting the test are - {out}")
+
+    mgr_host_object_list = []
+    for node in ceph_nodes:
+        if node.role == "mgr":
+            mgr_host_object_list.append(node)
+
+    mgr_daemon_list = mgr_obj.get_mgr_daemon_list()
+
+    log_lines = [
+        "mgr load Traceback",
+        "TypeError: __init__() got an unexpected keyword argument 'original_weight'",
+    ]
+
+    ceph_version = rados_obj.run_ceph_command(cmd="ceph version")
+    log.info(f"Current version on the cluster : {ceph_version}")
+    match_str = re.match(
+        r"ceph version (\d+)\.(\d+)\.(\d+)-(\d+)", ceph_version["version"]
+    )
+    major, minor, patch, build = match_str.groups()
+    bug_exists = False
+    exceptional_flag = False
+    if int(major) < 18:
+        bug_exists = True
+    elif int(major) == 18 and int(minor) < 2:
+        bug_exists = True
+    elif int(major) == 18 and int(minor) == 2 and int(patch) < 1:
+        bug_exists = True
+    elif int(major) == 18 and int(minor) == 2 and int(patch) == 1 and int(build) <= 234:
+        bug_exists = True
+
+    # Check the labels
+    cmd_host_ls = "ceph orch host ls"
+    out = rados_obj.run_ceph_command(cmd=cmd_host_ls)
+    drain_host = None
+    for node in out:
+        if "_no_schedule" in node["labels"]:
+            drain_host = node["hostname"]
+    if drain_host is None:
+        osd_hosts = rados_obj.get_osd_hosts()
+        log.info(f"The osd nodes list are-{osd_hosts}")
+        drain_host = random.choice(osd_hosts)
+
+    init_time, _ = installer.exec_command(cmd="sudo date '+%Y-%m-%d %H:%M:%S'")
+    log.info(f"The test execution  started at - {init_time} ")
+    osd_count_before_test = ""
+    try:
+        osd_count_before_test = get_node_osd_list(rados_obj, ceph_nodes, drain_host)
+        log.info(
+            f"The OSDs in the drain node before starting the test - {osd_count_before_test} "
+        )
+        mgr_daemon.start()
+        service_obj.remove_custom_host(host_node_name=drain_host)
+        time.sleep(300)
+        end_time, _ = installer.exec_command(cmd="sudo date '+%Y-%m-%d %H:%M:%S'")
+        log.info(f"The test execution  ends at - {end_time}")
+        if not verify_mgr_traceback_log(
```

Comment:
Copy this block to exception as well. 

In exception, error logs should be seen, and in try block, we should not see the error logs

Reply:
Added the check in the exception block.